src/Compilers/CSharp/Test/Symbol/Symbols/Source/FileModifierTests.cs

```diff
@@ -226,11 +414,210 @@ static void Main()
             Diagnostic(ErrorCode.ERR_NameNotInContext, "C").WithArguments("C").WithLocation(5, 9));
     }
 
+    [Fact]
+    public void Generic_01()
+    {
+        var source = """
+        using System;
+
+        C<int>.M(1);
+
+        file class C<T>
+        {
+            public static void M(T t) { Console.Write(t); }
+        }
+        """;
+
+        var verifier = CompileAndVerify(SyntaxFactory.ParseSyntaxTree(source, options: TestOptions.RegularPreview, path: "path/to/MyFile.cs", encoding: Encoding.Default), expectedOutput: "1", symbolValidator: symbolValidator);
+        verifier.VerifyDiagnostics();
+
+        verifier.VerifyIL("C<T>@MyFile.M(T)", @"
+{
+  // Code size       12 (0xc)
+  .maxstack  1
+  IL_0000:  ldarg.0
+  IL_0001:  box        ""T""
+  IL_0006:  call       ""void System.Console.Write(object)""
+  IL_000b:  ret
+}
+");
+
+        var comp = (CSharpCompilation)verifier.Compilation;
+        var c = comp.GetMember("C");
+        Assert.Equal("<MyFile>F0__C`1", c.MetadataName);
+
+        void symbolValidator(ModuleSymbol module)
+        {
+            Assert.Equal(new[] { "<Module>", "Program", "<MyFile>F0__C" }, module.GlobalNamespace.GetMembers().Select(m => m.Name));
+
+            // note: the arity is "unmangled" out of the name when loading metadata symbols.
+            // however, since file types aren't meant to be referenced outside the declaring file, we don't go to the effort of "unmangling" the file identifier out of the name.
+            var classC = module.GlobalNamespace.GetMember<NamedTypeSymbol>("<MyFile>F0__C");
+            Assert.Equal("<MyFile>F0__C`1", classC.MetadataName);
+            Assert.Equal(new[] { "M", ".ctor" }, classC.MemberNames);
+        }
+    }
```

Comment:
Do we need to account for the file name from a line directive?

Reply:
This would depend on how the language specification defines "file".

Reply:
https://github.com/dotnet/csharpstandard/blob/95e8d0ff3f5a85a8269173a8f5f3a3de70f5f342/standard/lexical-structure.md#658-line-directives

> Line directives may be used to alter the line numbers and compilation unit names that are reported by the compiler in output such as warnings and errors. These values are also used by caller-info attributes ([§21.5.5](attributes.md#2155-caller-info-attributes)).

The line directive causes the compiler to treat the compilation unit as having a different name. It doesn't cause it to treat it as being a different compilation unit.

I think it would be reasonable for us to use the name given by the line directive when determining what file name to use in the metadata name and in symbol display. However, I don't think it's required for correctness of the implementation. I'd like to file an issue to address it after merging the feature.

Reply:
I'm not sure I follow re correctness. Consider:
```C#
#line 1 "file1.cs"
file class C {}
#line 1 "file2.cs"
file class C {}
```

If the `#line` directive is used for scoping of file types the above is valid code as the classes are defined in different files. If he directive is ignored then the above is illegal as those declarations are not partial but defined in the same scope.

If both declarations were partial then both interpretations are valid however the former creates 2 types while the latter creates a single merged type.

Reply:
Sounds good. So `#file` does not affect scoping.